llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -2702,8 +2703,17 @@ void InnerLoopVectorizer::scalarizeInstruction(const Instruction *Instr,
   bool IsVoidRetTy = Instr->getType()->isVoidTy();
 
   Instruction *Cloned = Instr->clone();
-  if (!IsVoidRetTy)
+  if (!IsVoidRetTy) {
     Cloned->setName(Instr->getName() + ".cloned");
+#if !defined(NDEBUG)
+    // Verify that VPlan type inference results agree with the type of the
```

Comment:
Check should appear in execute, of every recipe (fine to add others later), but easier for replicate recipe to have it here, once?

Reply:
Yep it is simpler here, as the current structure of VPReplicateRecipe::execute uses earlier exits for different cases